std/experimental/allocator/package.d

```diff
@@ -1489,6 +1489,11 @@ private T[] uninitializedFillDefault(T)(T[] array) nothrow
             memset(array.ptr, 0xff, T.sizeof * array.length);
         return array;
     }
+    else static if (is(T == void))
+    {
+        // don't initialize void
+        return array;
+    }
```

Comment:
Just because it's `void[]` doesn't mean it shouldn't be initialised at all.

I think the correct change is to change `std.experimental.allocator.common.isInitAllZeroBits` to evaluate to `true` for `void` (and `void[N]`).

Reply:
Hmm, DMD does seem to initialize `void[]` to zeros - https://run.dlang.io/is/Q4lNNs